frontend/src/lib/k8s/KubeObject.ts

```diff
@@ -126,12 +125,16 @@ export class KubeObject<T extends KubeObjectInterface | KubeEvent = any> {
       namespace: this.getNamespace(),
       name: this.getName(),
     };
-    const link = createRouteURL(this.detailsRoute, params);
-    return link;
+    const cluster = this.cluster ?? getCluster();
+    if (this.isNamespaced) {
+      return `/c/${cluster}/${this._class().apiName}/${params.namespace}/${params.name}`;
+    }
+    return `/c/${cluster}/${this._class().apiName}/${params.name}`;
```

Comment:
Is replacing a function that produces a full URL by a repetition of the different URL parts a good idea? I'd say these changes make things harder to maintain and more error prone.

Reply:
yeah in the current form it will make it a bit worse. 
but if we take this convention a step further and replace all the routes with:
```js
{
details: {
   path: '/:apiName/:?namespace/:name',
   component: KubeObjectDetails,
},
list: {
  path: '/:apiName/',
  component: KubeObjectList
}
}
```

I think it will make it less error prone in the long term